modules/sdp/SDP.js

```diff
@@ -48,6 +51,39 @@ SDP.prototype.removeTcpCandidates = false;
  */
 SDP.prototype.removeUdpCandidates = false;
 
+/**
+ * Adds a new m-line for receiving source of the given media type.
+ *
+ * @param {Mediatype} mediaType media type of the new source that is being added.
+ */
+SDP.prototype.addMlineForNewLocalSource = function(mediaType) {
+    const mid = this.media.length;
```

Comment:
As mentioned in another comment, I think if we put source names as `mid` for the sender mlines that would make it easier to find transceivers and recycle the mlines.

Reply:
No I don't think this is a good idea. The same mids show up in both the local and remote description and these mids do not make sense when you are trying to generate new m-lines for new remote sources. They wouldn't follow a pattern and recycling them will not be possible.

Reply:
They would be dedicated to local tracks and can be recycled for that purpose, but again if you think it's easier this way then that's ok. Still I think it will be more error prone. The `if` conditions that rely on ordering, finding media type and inactive or recvonly is hard to understand and follow without tons of comments.

Reply:
The idea is to use the first available sender when you are adding/replacing it on the peerconnection. I still don't get what you mean by dedicated m-lines. Having one fixed per JitsiLocalTrack means there will be many unused ones since every device change for a given source is a new JitsiLocalTrack and will be a new m-line.